### PR TITLE
Adds a validation for plugins script to check if java is set

### LINKED ISSUE #13613: plugin script doesn't detect when Java is not installed
When JAVA is not installed, `bin/elasticsearch` outputs:

```
Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME
```

But when you run `bin/plugin`, you get:

```
bin/plugin: 1: eval: -client: not found
```

### PR DESCRIPTION
**Pull request by andrestc**
This does the same validation as in elasticearch bin script.
Closes #13613

### CODE REVIEW

**Comment by nik9000** [comment]
Looks good to me! I tested it like so:
```bash
$ export JAVA=$(which java)
$ sudo chmod -x $JAVA
$ bin/plugin
Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME
$ bin/elasticsearch
Could not find any executable java binary. Please install java in your PATH or set JAVA_HOME
$ sudo chmod +x $JAVA
$ bin/plugin
ERROR: command not specified
<snip>
$ bin/elasticsearch
[2015-09-17 14:07:28,822][INFO ][org.elasticsearch.node   ] [Tom Cassidy] version[3.0.0-SNAPSHOT], 
<snip>
^C
```

How would you feel about adding a test like that to the tests in `qa/vagrant`? There are instructions for running them in TESTING.asciidoc. I figure it'd be good to add this to [20_tar_package.bats](https://github.com/elastic/elasticsearch/blob/master/qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats).

The test in qa/vagrant are always run in a virtual machine so you can do mean stuff like `chmod -x /usr/bin/java` without worrying about upsetting someone.

**Comment by andrestc** [comment]
Sure @nik9000, i'll definitely add the test! I wanted to add it to the original PR but couldn't find where to place them. I'll update this once im done.

**Comment by andrestc** [comment]
@nik9000 I added 2 separated test cases, one for each script. Took a while to download everything I needed for the tests. Had to also read a bit on bats.

**Comment by nik9000** [comment]
Sorry!  Vagrant is lots of downloading. Did you do just one box or all of
them?  One box is fine. Certainly for this. All of them is a really odyssey
of downloading.

**Comment by andrestc** [comment]
Its ok! I ended up downloading all of them but ran tests on a single one.

File: qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats [hunk 1]
```diff
@@ -72,6 +72,17 @@ setup() {
     verify_archive_installation
 }
 
+@test "[TAR] elasticsearch fails if java executable is not found" {
+  export JAVA=$(which java)
```
Comment (nik9000) [inline qa/vagrant/src/test/resources/packaging/scripts/20_tar_package.bats]: s/export/local/ maybe. Not a big deal though.

**Comment by andrestc** [comment]
I addressed your comment and updated the branch, @nik9000.
Thanks.

**Comment by nik9000** [comment]
It looks like this fails on centos-7!  Maybe just check that the string is _in_ the output rather than that the entire output is the string?

Other than that its good!

You should probably try something like `mvn -pl qa/vagrant -Dtests.vagrant=all -DtestScripts='20_*' verify` to verify the new test on all the VMs we support. Sorry about that. Its certainly a pain that all the distros are different.

**Comment by andrestc** [comment]
It is ok, I should have tested on all before. My bad.

I'm pretty sure its fixed now. I tested in all boxes except for fedora-22 because I was getting the following output when provisioning the box:
```
==> fedora-22: Transaction Summary
==> fedora-22: ================================================================================
==> fedora-22: Install  38 Packages
==> fedora-22: Total download size: 55 M
==> fedora-22: Installed size: 193 M
==> fedora-22: Downloading Packages:
==> fedora-22: [Errno unknown] Downloading from http://fedora.c3sl.ufpr.br/linux/releases/22/Everything/x86_64/os/Packages/l/libX11-common-1.6.3-1.fc22.noarch.rpm
was successful but error encountered while checksuming: read(19) failed: 
Protocol error: u'Input/Output error'
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
```
I'm not sure what maybe causing this. I suspect that, if others aren't having this problem, maybe its related to the mirror its using.

Also, I noted Wheezy aka Debian 7 is not on tests anymore. I think we should remove it from docs. If so, i can do it if you want, should I open a new PR?

Thanks for the review!

**Comment by andrestc** [comment]
Actually after retrying a few more times I managed to provision fedora-22. Tests passed fine on it too!

**Comment by nik9000** [comment]
> Also, I noted Wheezy aka Debian 7 is not on tests anymore. I think we should remove it from docs. If so, i can do it if you want, should I open a new PR?

Oh! A new pr would be wonderful.

**Comment by nik9000** [comment]
> Actually after retrying a few more times I managed to provision fedora-22. Tests passed fine on it too!

OK! I got everything passing locally as well! I'm going to merge this then.

**Comment by nik9000** [comment]
Merge to 2.x and master. Thans @andrestc!

**Pull request merged**